### PR TITLE
Add mechanism to exit scene and show message for closed or full rooms

### PR DESCRIPTION
**Pull request by gfodor**
Goes with https://github.com/mozilla/reticulum/pull/18

This PR:

- Pushes a bunch of functions in `hub.js` into local bindings in `onReady`
- Adds logic to show message + exit when a hub is full or closed

### CODE REVIEW

File: src/hub.js [hunk 1]
```diff
@@ -255,19 +148,135 @@ function mountUI(scene, props = {}) {
...
+
+    if (!qsTruthy("offline")) {
+      document.body.addEventListener("connected", () => {
+        hubChannel.sendEntryEvent().then(() => {
+          store.update({ lastEnteredAt: moment().toJSON() });
+        });
+      });
+
+      scene.components["networked-scene"].connect().catch(connectError => {
+        // hacky until we get return codes
```
Comment (gfodor) [inline src/hub.js]: actual change is here, other code is largely just moved around

Comment (netpro2k) [reply src/hub.js]: Looks like the actual text is "Room is full." https://github.com/mozilla/janus-plugin-sfu/blob/master/src/lib.rs#L440

File: src/react-components/ui-root.js [hunk 2]
```diff
@@ -499,33 +500,35 @@ class UIRoot extends Component {
...
       return (
         <IntlProvider locale={lang} messages={messages}>
-          <div className="loading-panel">
-            <div className="loader-wrap">
-              <div className="loader">
-                <div className="loader-center" />
-              </div>
-            </div>
+          <div className="exited-panel">
             <div className="loading-panel__title">
```
Comment (netpro2k) [inline src/react-components/ui-root.js]: Are these supposed to be `exited-panel` or are we just re-using the loading-panel styles? I think for proper BEM you should alias them in the css, but this is super minor.

File: src/hub.js (hunk 1 above)
Comment (mquander) [reply src/hub.js]: brb, putting in an error that says "network connection too full, can't send packets"

Comment (brianpeiris) [reply src/hub.js]: Worth noting this is dependent on https://github.com/mozilla/naf-janus-adapter/pull/31

Comment (brianpeiris) [reply src/hub.js]: Also dependent on https://github.com/mozilla/mr-social-client/pull/207

File: src/react-components/ui-root.js (hunk 2 above)
Comment (gfodor) [reply src/react-components/ui-root.js]: I wrote a separate bug for this, will deal with it at some point

**Pull request merged**